### ISSUE TITLE
[BUG] Z-normalisation numba function produces wrong results with integer inputs

### ISSUE THREAD

**Issue by baraline** [aeon-toolkit/aeon#2250]
### Describe the bug

Functions such as `z_normalise_series_2d` from `utils\numba\general.py` when used with integer array as inputs ouput an integer array which do not correspond to expected normalization results. The cause is that these functions are using `arr = np.zeros_like(X)` to create the returned array, which will use the same dtype as the input X.

Replacing this with `arr = np.zeros(X.shape)` should fix the issue.

### Steps/Code to reproduce the bug

```
from aeon.utils.numba.general import z_normalise_series_2d
z_normalise_series_2d(np.array([[1,2,3]],dtype=int))
```

### Expected results

Out[64]: array([[-1.22474487,  0.        ,  1.22474487]])

### Actual results

Out[65]: array([[-1,  0,  1]])

**Comment by aryan0931** [comment]
hello , I would like to work on this issue.

**Comment by baraline** [comment]
Hey, sorry, just seen your message, this issue should be fixed with #2176.
